configure

```diff
@@ -20,9 +20,37 @@ else
   EXTRA_PKG_LIBS=-latomic
 fi
 
+# Detect whether it requires c++17
+echo "#include <cstdio>
+#include <ctime>
+// Example adapted from https://en.cppreference.com/w/cpp/chrono/c/timespec_get
+
+namespace std {
+
+	int main()
+	{
+		timespec ts;
+		timespec_get(&ts, TIME_UTC);
+		char buf[100];
+		strftime(buf, sizeof buf, "%D %T", gmtime(&ts.tv_sec));
+		printf("Current time: %s.%09ld UTC\n", buf, ts.tv_nsec);
+		return 0;
+	}
+
+}
+" | ${CXX} -x c++ -std=c++11 - -o /dev/null > /dev/null 2>&1
```

Comment:
Note that there are some platforms that don't support `std-=c++11`. I'm not even sure that some of the more obscure platforms (like the compiler on Solaris) have the `-std` option